php/Terminus/Commands/SiteCommand.php

```diff
@@ -766,6 +754,55 @@ public function hostnames($args, $assoc_args) {
           )
         );
           break;
+      case 'lookup':
+        $this->log()->warning('This operation may take a long time to run.');
+        $hostname  = Input::string(
+          $assoc_args,
+          'hostname',
+          'Please enter a hostname to look up.'
+        );
+        $sites    = $this->sites->all();
+        $data     = null;
+        foreach ($sites as $site_id => $site) {
+          $environments = array('dev', 'test', 'live');
+          foreach ($environments as $env_name) {
+            $environment = $site->environments->get($env_name);
+            $hostnames   = array_keys((array)$environment->getHostnames());
+            if (in_array($hostname, $hostnames)) {
```

Comment:
Could we move this logic into the SitesCollection?  So we could call `$this->sites->findByHostname('something.com')` or something. 

Thought that would be weird cause it would return an environment rather than an entire site. Yeah, maybe this is fine as is.